src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/Processors/LedgerTransactionMarkers/ManifestProcessor.cs

```diff
@@ -108,128 +108,169 @@ public void OnTransactionScan(CoreModel.CommittedTransaction transaction, long s
                 {
-                    _ledgerTransactionMarkersToAdd.Add(
-                        new ManifestAddressLedgerTransactionMarker
-                        {
-                            Id = _context.Sequences.LedgerTransactionMarkerSequence++,
-                            StateVersion = stateVersion,
-                            OperationType = LedgerTransactionMarkerOperationType.BadgePresented,
-                            EntityId = re.DatabaseId,
-                        });
+                    throw new UnreachableException($"Entity: {proofResourceAddress} was not present in referenced entities dictionary.");
                 }
+
+                ledgerTransactionMarkersToAdd.Add(
+                    new ManifestAddressLedgerTransactionMarker
+                    {
+                        Id = _context.Sequences.LedgerTransactionMarkerSequence++,
+                        StateVersion = stateVersion,
+                        OperationType = LedgerTransactionMarkerOperationType.BadgePresented,
+                        EntityId = referencedEntity.DatabaseId,
+                    });
             }
 
-            foreach (var address in extractedAddresses.ResourceAddresses)
+            foreach (var resourceAddress in extractedAddresses.ResourceAddresses)
             {
-                if (_referencedEntities.TryGet(address, out var re))
+                if (!_referencedEntities.TryGet(resourceAddress, out var referencedEntity))
                 {
-                    _ledgerTransactionMarkersToAdd.Add(
-                        new ManifestAddressLedgerTransactionMarker
-                        {
-                            Id = _context.Sequences.LedgerTransactionMarkerSequence++,
-                            StateVersion = stateVersion,
-                            OperationType = LedgerTransactionMarkerOperationType.ResourceInUse,
-                            EntityId = re.DatabaseId,
-                        });
+                    throw new UnreachableException($"Entity: {resourceAddress} was not present in referenced entities dictionary.");
                 }
+
+                ledgerTransactionMarkersToAdd.Add(
+                    new ManifestAddressLedgerTransactionMarker
+                    {
+                        Id = _context.Sequences.LedgerTransactionMarkerSequence++,
+                        StateVersion = stateVersion,
+                        OperationType = LedgerTransactionMarkerOperationType.ResourceInUse,
+                        EntityId = referencedEntity.DatabaseId,
+                    });
             }
 
-            foreach (var address in extractedAddresses.AccountsRequiringAuth)
+            foreach (var addressWithEntityType in extractedAddresses.AccountsRequiringAuth)
             {
-                if (_referencedEntities.TryGet(address, out var re))
+                if (_referencedEntities.TryGet(addressWithEntityType.Address, out var referencedEntity))
```

Comment:
I'm a bit confused. It looks like `_referencedEntities` just extracts all addresses from the manifest, but at this stage, we haven't checked that these addresses either exist on ledger already, right? (or were created in this transaction?)

Or does this happen _after_ the resolution of the `_referencedEntities`? i.e. at this point, we know that all `_referencedEntities` either exist on ledger already, or were created in this transaction (or previous transaction in the batch)? What happens at resolution time if `_referencedEntities` don't exist (and aren't preallocated addresses)? I assume that we already throw?

Either way, I don't think the check here really makes sense? Perhaps we should discuss on a call?

Reply:
Following conversations with Pawel:
* At this point, we have done a DB load of the `MarkSeenAddress` addresses
* We _don't throw_ if an address doesn't match - we just don't see it in the `TryGet` method.

Reply:
So this looks good 👍